src/ipoplib.py

```diff
@@ -626,21 +747,46 @@ def parse_config():
     if not ("xmpp_username" in CONFIG and "xmpp_host" in CONFIG):
         raise ValueError("At least 'xmpp_username' and 'xmpp_host' must be "
                          "specified in config file or string")
-
+    # Use custom keyring incase of non-interactive environment
+    # (uses a fixed password)
+    if isinstance(keyring.get_keyring(),keyring.backends.file.EncryptedKeyring):
```

Comment:
plaintext keyring module has priority of 0.5, encryptedkeyring has 0.6. But encrypted keyring module requires pycrypto and json module to be installed. So when the library is imported, if these(pycrypto and json) modules are not present, encryptedkeyring module will not be used as backend and plaintextkeyring is chosen. Other keyrings have much higher priorities and are chosen if they are available(determined based on system variables like XDG_CURRENT_DESKTOP and DBUS_SESSION_BUS_ADDRESS).